src/btree/bt_slvg.c

```diff
@@ -1017,15 +997,21 @@ __slvg_col_range_overlap(
 	 * Split a_trk into two parts, the key range before b_trk and the
 	 * key range after b_trk.
 	 */
-	WT_RET(__slvg_trk_split(session, a_trk, &new));
+	WT_RET(__wt_calloc_one(session, &new));
+	new->shared = a_trk->shared;
+	new->ss = a_trk->ss;
+	++a_trk->shared->ref;
 
 	/*
 	 * Second, reallocate the array of pages if necessary, and then insert
 	 * the new element into the array after the existing element (that's
 	 * probably wrong, but we'll fix it up in a second).
 	 */
-	WT_RET(__wt_realloc_def(
-	    session, &ss->pages_allocated, ss->pages_next + 1, &ss->pages));
+	if ((ret = __wt_realloc_def(session,
+	    &ss->pages_allocated, ss->pages_next + 1, &ss->pages)) != 0) {
+		__wt_free(session, new);
```

Comment:
Based on what you pointed out in your last comment, don't we need to decrement `a_trk->shared->ref` on the error paths too?